polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -5251,7 +5277,11 @@ bool MLIRASTConsumer::HandleTopLevelDecl(DeclGroupRef dg) {
         emitIfFound.count(name) || fd->hasAttr<OpenCLKernelAttr>() ||
         fd->hasAttr<SYCLDeviceAttr>()) {
       // Functions are created in the host context by default.
-      functionsToEmit.emplace_back(*fd, FunctionContext::Host);
+      FunctionToEmit F{*fd, FunctionContext::Host};
+      LLVM_DEBUG(llvm::dbgs()
+                 << __LINE__ << ": Pushing " << F.context << " function "
+                 << fd->getNameAsString() << " to functionsToEmit\n");
+      functionsToEmit.emplace_back(F);
```

Comment:
No need to change `emplace_back` call.

Reply:
emplace_back is not needed because there is nothing to construct.

Reply:
I'm proposing not creating the instance in line 5280 and creating it in place.

Reply:
I need the instance though because I referenced at line 5282

Reply:
If we decide to create the instance at line 5280, 
```suggestion
      functionsToEmit.push_back(F);
```

Reply:
done